ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/ethernet-interfaces.j2

```diff
@@ -74,10 +74,16 @@ interface {{ ethernet_interface.name }}
 {%             endif %}
 {%         endif %}
 {%         if ethernet_interface.mode is arista.avd.defined('trunk') %}
-{%             if ethernet_interface.native_vlan is arista.avd.defined %}
+{%             if ethernet_interface.native_vlan_tag is arista.avd.defined(true) %}
+   switchport trunk native vlan tag
+{# next is a way of checking the type is int only if the value is defined #}
+{# not doing the two checks would lead to false warnings when the value is undefined #}
+{%             elif ethernet_interface.native_vlan is arista.avd.defined and
+                    ethernet_interface.native_vlan is arista.avd.defined(fail_action='warning', var_type='int', var_name='native_vlan') %}
```

Comment:
I did this on purpose - otherwise everytime it is not defined (which can be on purpose) then you get an error message. By testing if it exists first and then that it has correct type it gives the behavior I want.

This could be fixed if we added an option is `defined` when we check the type - or added some way to return if the type is correct only when the variable is defined. 

@ClausHolbechArista  is this possible?

Reply:
when you set `fail_action="warning"` it will warn if the variable is undefined _or_ if the `var_type` is not matching. We cannot warn differently.

Reply:
So I think the dual checks is best for now. I don't think we need to add this warning for non-integer, since we will soon get input validation covering all this.

Reply:
I am happy to remove the double check if this is preferred - just thought it would give users some advance warnings about the change and what is available

Reply:
Addressed in 9ec01a2